### PR TITLE
ncdu@1 1.18.1 (new formula)

### PR DESCRIPTION
**Pull request by OctopusET**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
ncdu version 1 is still supported and it's useful when you don't want to use or can't use zig.
Ref: https://dev.yorhel.nl/ncdu/changes

### CODE REVIEW

**Comment by OctopusET** [comment]
What would be better?
Change binary name to `ncdu1`
or
Make it conflict with ncdu

**Review by carlocab — CHANGES_REQUESTED** [review]
Not convinced this is really needed, since you don't need to install Zig to use `ncdu` if you are in a supported configuration.

File: Formula/ncdu1.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+    system "autoreconf", "-i" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/ncdu -v")
```
[the remark this answers is not in the document]
Comment (OctopusET) [reply Formula/ncdu1.rb:25]: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ncdu.rb
This formula file is actually from the last ncdu1 version.
I think I can use test from ncdu2 formula.

File: Formula/ncdu1.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class Ncdu1 < Formula
```
[the remark this answers is not in the document]
Comment (OctopusET) [reply Formula/ncdu1.rb]: done

File: Formula/ncdu1.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (OctopusET) [reply Formula/ncdu1.rb]: Imported test from ncdu2

**Comment by carlocab** [comment]
> ncdu version 2 can be considered unstable because of zig.
> 
> The developer has stated this (https://dev.yorhel.nl/doc/ncdu2), and the current formula we use includes a patch because it is not buildable out of the box. It is possible that in the future, ncdu2 will have issues with zig or that updates will be withheld when a patch is needed.  Since ncdu1 does not have stability issues, it would be helpful to add this to increase stability.

The latest `ncdu` is buildable out of the box. The patch is to enable packaging using `brew`.

In any case, the lack of stability of Zig doesn't seem like a big issue at the moment, especially since `ncdu` seems to be careful about making sure it builds with recent tagged releases of Zig.

We can consider adding this if we run into trouble updating `ncdu` using Zig, but right now this seems premature.

**Comment by OctopusET** [comment]
Reasonable! but if there's problem with zig on supported system someday, I'd love to contribute.

Still I and there are some people need this. Because they use unsupported version (e.g Catalina) So, I'd like to maintain my own [taps](https://github.com/OctopusET/homebrew-42).


btw, should I close this PR?

**Comment by carlocab** [comment]
> Reasonable! but if there's problem with zig on supported system someday, I'd love to contribute.

Sure, please do.

> Still I and there are some people need this. Because they use unsupported version (e.g Catalina) So, I'd like to maintain my own [taps](https://github.com/OctopusET/homebrew-42).

That's a reasonable use-case, but we no longer actively support Catalina. Thanks for the PR anyway!

**Pull request closed without merging**